### PR TITLE
Fee payment

### PR DESCRIPTION
**Pull request by KarishmaBothara**

### CODE REVIEW

File: packages/api/package.json [hunk 1]
```diff
@@ -31,9 +31,7 @@
     "eventemitter3": "^4.0.0"
   },
   "devDependencies": {
-    "@cennznet/cennznut": "^1.0.0",
```
Comment (holygits) [inline packages/api/package.json]: Thanks for fixing this

File: packages/api/test/e2e/tx.e2e.ts [hunk 2]
```diff
@@ -16,11 +16,16 @@
...
 
-import {AssetId, AssetOptions} from '@cennznet/types';
+import {AssetId, AssetOptions, FeeExchangeV1} from '@cennznet/types';
 import {SimpleKeyring, Wallet} from '@cennznet/wallet';
 import {SubmittableResult} from '@polkadot/api';
 import {cryptoWaitReady} from '@plugnet/util-crypto';
 import initApiPromise from '../../../../jest/initApiPromise';
+import {ChargeTransactionPayment, FeeExchange} from '@cennznet/types/runtime/transaction-payment';
+import {Enum, Option, getTypeRegistry as registry} from '@polkadot/types';
+// import {getTypeRegistry} from '@polkadot/types';
```
Comment (holygits) [inline packages/api/test/e2e/tx.e2e.ts]: Can we remove this?

File: packages/types/src/extrinsic/SignerPayload.ts [hunk 3]
```diff
@@ -2,115 +2,125 @@
...
+  tip: Compact<Balance>;
+  version: u8;
+  doughnut: Option<Doughnut>;
+  transactionPayment: ChargeTransactionPayment;
 }
 
 export interface SignerPayloadJSON extends SignerPayloadJSONBase {
-    doughnut?: string;
+  doughnut?: string;
+  //  transactionPayment?: {
```
Comment (holygits) [inline packages/types/src/extrinsic/SignerPayload.ts]: Can we remove this?

File: packages/types/src/extrinsic/types.ts [hunk 4]
```diff
@@ -35,7 +35,8 @@ export interface ExtrinsicOptions {
 export interface ExtrinsicSignatureOptions {
   isSigned?: boolean;
   doughnut?: Doughnut;
-  feeExchange?: any;
+  transactionPayment?: ChargeTransactionPayment;
+  //  feeExchange?: any;
```
Comment (holygits) [inline packages/types/src/extrinsic/types.ts]: Can we remove this?

File: packages/types/src/runtime/transaction-payment/index.ts [hunk 5]
```diff
@@ -15,21 +15,51 @@
...
+import {AnyJson, AnyJsonObject} from '@polkadot/types/types';
 
 /* [[FeeExchangeV1]] when included in a transaction it indicates network fees should be
  * paid in `assetId` by paying up to `maxPayment` after the exchange rate is calculated.
  */
 export class FeeExchangeV1 extends Struct {
-  constructor(value: any) {
-    super({assetId: 'AssetId', maxPayment: 'Compact<Balance>'}, value);
+  constructor(value?: any) {
+    super({assetId: 'Compact<AssetId>', maxPayment: 'Compact<Balance>'}, value);
```
Comment (holygits) [inline packages/types/src/runtime/transaction-payment/index.ts]: 👍

File: packages/api/src/types.ts [hunk 6]
```diff
@@ -48,117 +49,118 @@ export * from '@polkadot/api/types';
...
+  sdkClass?: Constructor<any>;
+  sdkRxClass?: Constructor<any>;
+  types?: RegistryTypes;
+  derives?: DeriveCustom;
 }
 
 export interface SignerOptions extends SignerOptionsBase {
-    doughnut?: DoughnutValue;
-    feeExchange?: FeeExchangeValue;
+  // doughnut?: DoughnutValue;
```
Comment (holygits) [inline packages/api/src/types.ts]: Can we remove `feeExchange` here?
Leave doughnut though

File: packages/api/test/e2e/tx.e2e.ts [hunk 7]
```diff
@@ -144,21 +185,25 @@ describe('e2e transactions', () => {
...
+        const transactionPayment = {
+          tip: 2,
+          feeExchange:  {
+            FeeExchangeV1: feeExchange,
           },
         };
-        return tx.signAndSend(sender.address, txOpt, ({events, status}) => {
+
+        const tx = api.tx.genericAsset.transfer(16001, receiver.address, 100);
+        return tx.signAndSend(sender.address, {transactionPayment}, ({events, status}) => {
```
Comment (holygits) [inline packages/api/test/e2e/tx.e2e.ts]: Does it work with `sender.Keypair` aswell?

Comment (KarishmaBothara) [reply packages/api/test/e2e/tx.e2e.ts]: yes

File: packages/api/test/e2e/tx.e2e.ts [hunk 8]
```diff
@@ -130,12 +166,17 @@ describe('e2e transactions', () => {
...
-          .transfer(16000, receiver.address, 10000)
-          .signAndSend(senderKeypair, {feeExchange}, ({events, status}) => {
+        const feeExchangeV1 = {FeeExchangeV1 : feeExchange};
+        const transactionPayment = {
+          tip: '2', feeExchange:  feeExchangeV1,
+        };
+
+         api.tx.genericAsset
+          .transfer(16001, receiver.address, 100)
+          .signAndSend(senderKeypair, {transactionPayment}, ({events, status}) => {
```
Comment (KarishmaBothara) [inline packages/api/test/e2e/tx.e2e.ts]: This one is with key pair

File: packages/api/src/types.ts (hunk 6 above)
Comment (KarishmaBothara) [reply packages/api/src/types.ts]: yes..will do

File: packages/api/test/e2e/tx.e2e.ts (hunk 2 above)
Comment (KarishmaBothara) [reply packages/api/test/e2e/tx.e2e.ts]: yes..will remove all unwanted imports

File: packages/types/src/extrinsic/SignerPayload.ts (hunk 3 above)
Comment (KarishmaBothara) [reply packages/types/src/extrinsic/SignerPayload.ts]: sorry.. should have removed earlier...

**Pull request merged**